core/arch/arm/kernel/secure_partition.c

```diff
@@ -134,22 +134,55 @@ bool sp_has_exclusive_access(struct sp_mem_map_region *mem,
 	return !sp_mem_is_shared(mem);
 }
 
-static void sp_init_info(struct sp_ctx *ctx, struct thread_smc_args *args)
+static TEE_Result sp_init_info(struct sp_ctx *ctx, struct thread_smc_args *args,
+			       const void * const fdt, vaddr_t *va,
+			       size_t *num_pgs)
 {
 	struct sp_ffa_init_info *info = NULL;
+	int nvp_count = 1;
+	size_t nvp_size = sizeof(struct sp_name_value_pair) * nvp_count;
+	size_t info_size = sizeof(*info) + nvp_size;
+	size_t fdt_size = fdt_totalsize(fdt);
+	TEE_Result res = TEE_SUCCESS;
+	uint32_t perm = TEE_MATTR_URW | TEE_MATTR_PRW;
+	struct fobj *fo = NULL;
+	struct mobj *m = NULL;
+	static const char fdt_name[16] = "TYPE_DT\0\0\0\0\0\0\0\0";
```

Comment:
doesn't the below produce the same result (for the 16 cells)?
```c
	static const char fdt_name[16] = "TYPE_DT";
```

Reply:
It would. It's done to make it explicit that the array is zeroed out and that in the SP we can do a memcmp() instead of a strcmp().

Reply:
Ok, fine as is.
@jforissier, won't `static const char fdt_name[16] = "TYPE_DT";` also fill the remaining unset cells to 0?